controllers/storagecluster/provider_server.go

```diff
@@ -496,9 +496,12 @@ func (o *ocsProviderServer) createJob(r *StorageClusterReconciler, instance *ocs
 
 	if errors.IsNotFound(err) {
 		onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
-		err = r.Client.Create(context.Background(), onboardingSecretGeneratorJob)
-	}
-	if err != nil {
+		jobErr := r.Client.Create(context.Background(), onboardingSecretGeneratorJob)
+		if jobErr != nil && !errors.IsAlreadyExists(jobErr) {
+			r.Log.Error(jobErr, "failed to create onboarding job")
+			return reconcile.Result{}, jobErr
+		}
+	} else if err != nil {
```

Comment:
We want to fail if it already exists as we don't have guarantees about the state and content of the secrets when the job was originally invoked. 

Before this change, the reconciliation will fail and the next reconciliation will either create the job or fail again. Eventually, a job will be created where we have exact context and guarantees of its outcome.

Reply:
So, really, the "happy path" behavior is `errors.IsNotFound(err)` and we should have an error if `err == nil`?

Reply:
I believe I ran into this while I was hacking on the Job and it had gone into a crash loop. Does this mean that if a version of the Job was already created, we want to keep failing the reconcile loop until either the garbage collector or an admin deletes the extant Job?

Reply:
The retention time of the job is set to 0, so it should delete itself immediately after completion. 
I acknowledge that this approach is also lacking. We are looking at alternatives, making sure that the job itself will recognize the need to iterate again without involving a full new reconcile for storagecluster. But that is out of the scope of this PR.

Reply:
Okay, I can take this out of the PR and hold on to it in my dev branches until a different solution presents itself. If there's a place where this discussion happening, feel free to loop me in on it.

Reply:
Done.